src/backend/access/transam/varsup.c

```diff
@@ -514,6 +520,76 @@ ForceTransactionIdLimitUpdate(void)
 	return false;
 }
 
+Oid
+GetNewTempObjectId(void)
```

Comment:
we should add functionality to extension via hooks wherever possible, this will reduce the chances of regressing engine code